tools/depcheck/pkg/cmd/trace/filter.go

```diff
@@ -0,0 +1,143 @@
+package trace
+
+import (
+	"strings"
+
+	"github.com/gonum/graph"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+// FilterVendorPackages receives a graph and a set of vendorRootNames contained within the graph.
+// Returns a new graph with the sub-tree for each vendorRoot collapsed into just the vendorRoot.
+// Relationships among vendorRoots are kept: if a transitive closure exists between two
```

Comment:
Pretty sure that transitive closure refers to the set of all reachable nodes, not a navigable path.

In any case, future readers will probably prefer: "edges originating from vendorRoot subpackages are rewritten to originate from the vendorRoot and edges terminating at vendorRoot subpackages are rewritten to terminate at the vendorRoot"